### PR TITLE
WRR-6385: Converted sandstone/TimePicker/TimePickerBase HourPicker to functional component

### PR DESCRIPTION
**Pull request by paul-beldean-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Converted sandstone/TimePicker/TimePickerBase HourPicker to functional component.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRR-6385

### Comments

Enact-DCO-1.0-Signed-off-by: Paul Beldean paul.beldean@lgepartner.com

### CODE REVIEW

File: TimePicker/TimePickerBase.js [hunk 1]
```diff
@@ -27,45 +27,33 @@ const hours12 = [
...
-		if (state.prevValue !== props.value) {
-			const hours = props.hasMeridiem ? hours12 : hours24;
+	useEffect(() => {
+		const hoursArray = props.hasMeridiem ? hours12 : hours24;
 
-			return {
-				noAnimation: hours[state.prevValue] === hours[props.value],
-				prevValue: props.value
-			};
+		if (props.value !== prevValue) {
```
Comment (adrian-cocoara-lgp) [inline TimePicker/TimePickerBase.js:37]: let's keep the original order in `if` condition
```suggestion
		if (prevValue !== props.value) {
```

File: TimePicker/TimePickerBase.js [hunk 2]
```diff
@@ -27,45 +27,33 @@ const hours12 = [
...
-		const hours = hasMeridiem ? hours12 : hours24;
-
-		return (
-			<DateComponentPicker {...rest} noAnimation={this.state.noAnimation}>
-				{hours}
-			</DateComponentPicker>
-		);
-	}
-}
+	}, [prevValue, props.hasMeridiem, props.value]);
```
Comment (adrian-cocoara-lgp) [inline TimePicker/TimePickerBase.js:41]: please change dep array like so
```suggestion
	}, [prevValue, props]);
```

**Review by adrian-cocoara-lgp — APPROVED** [review]
LGTM 👍

File: TimePicker/TimePickerBase.js [hunk 3]
```diff
@@ -27,45 +27,33 @@ const hours12 = [
...
-		const hours = hasMeridiem ? hours12 : hours24;
-
-		return (
-			<DateComponentPicker {...rest} noAnimation={this.state.noAnimation}>
-				{hours}
-			</DateComponentPicker>
-		);
-	}
-}
+	}, [prevValue, props]);
```
Comment (juwonjeong) [inline TimePicker/TimePickerBase.js:41]:
```suggestion
	}, [hasMeridiem, prevValue, value]);
```

File: TimePicker/TimePickerBase.js [hunk 4]
```diff
@@ -27,45 +27,33 @@ const hours12 = [
...
+
+	const {hasMeridiem, ...rest} = props;
+	const hours = hasMeridiem ? hours12 : hours24;
+
+	return (
+		<DateComponentPicker {...rest} noAnimation={noAnimation}>
```
Comment (juwonjeong) [inline TimePicker/TimePickerBase.js:47]:
```suggestion
		<DateComponentPicker {...rest} noAnimation={noAnimation} value={value}>
```

**Review by juwonjeong — APPROVED** [review]
LGTM

**Pull request merged**